src/poetry/utils/env.py

```diff
@@ -533,26 +535,48 @@ def _full_python_path(self, python: str) -> str:
 
         return executable
 
-    def _detect_active_python(self, io: IO) -> str | None:
+    def _detect_active_python(self) -> str | None:
         executable = None
 
         try:
-            io.write_error_line(
+            self._io.write_error_line(
                 "Trying to detect current active python executable as specified in the"
                 " config.",
                 verbosity=Verbosity.VERBOSE,
             )
             executable = self._full_python_path("python")
-            io.write_error_line(f"Found: {executable}", verbosity=Verbosity.VERBOSE)
+            self._io.write_error_line(
+                f"Found: {executable}", verbosity=Verbosity.VERBOSE
+            )
         except CalledProcessError:
-            io.write_error_line(
+            self._io.write_error_line(
                 "Unable to detect the current active python executable. Falling back to"
                 " default.",
                 verbosity=Verbosity.VERBOSE,
             )
         return executable
 
-    def activate(self, python: str, io: IO) -> Env:
+    def _get_python_version(self) -> list[int]:
+        version_info = list(sys.version_info[:3])
+
+        if self._poetry.config.get("virtualenvs.prefer-active-python"):
+            executable = self._detect_active_python()
+
+            if executable:
+                python_patch = decode(
+                    subprocess.check_output(
+                        list_to_shell_command(
+                            [executable, "-c", GET_PYTHON_VERSION_ONELINER]
+                        ),
+                        shell=True,
+                    ).strip()
+                )
+
+                version_info = [int(v) for v in python_patch.split(".")[:3]]
+
+        return version_info
```

Comment:
```suggestion
    def _get_python_version(self) -> tuple[int, int, int]:
        if self._poetry.config.get("virtualenvs.prefer-active-python"):
            executable = self._detect_active_python()

            if executable:
                python_patch = decode(
                    subprocess.check_output(
                        list_to_shell_command(
                            [executable, "-c", GET_PYTHON_VERSION_ONELINER]
                        ),
                        shell=True,
                    ).strip()
                )

                return tuple(int(v) for v in python_patch.split(".")[:3])

        return sys.version_info[:3]
```
Using a tuple would more clearly indicate that we expect 3 items.

Reply:
I implemented a version that returns a tuple. `mypy` was a little bit in the way here ...